core/TypePtr.h

```diff
@@ -94,13 +107,13 @@ class TypePtr final {
         return static_cast<tagged_storage>(tag);
     }
 
-    static tagged_storage tagValue(Tag tag, uint32_t inlinedValue) {
+    static tagged_storage tagValue(Tag tag, uint64_t inlinedValue) {
```

Comment:
Why does the `inlinedValue` go up to 64 bits? Don't we only have `64 - 16 - 1 = 47` bits to spare here?

Reply:
We do, but variable-width integer types haven't made it to C++ yet (there's no `int47_t`).  (Note that we do at least have an `ENFORCE` that limits us to 48-bit values.)